### PR TITLE
 Add support for vertical alignment of styled paragraphs inside table cells

### PR DESCRIPTION
**Pull request by adrg**

### CODE REVIEW

File: creator/styled_paragraph_test.go [hunk 1]
```diff
@@ -798,3 +798,69 @@ func TestStyledLinkRotation(t *testing.T) {
...
+			}
+
+			for i := 4; i <= 20; i += 2 {
+				createTable(c, chunk, alignment.Alignment, float64(i))
+			}
+		}
+	}
+
+	// Write output file.
+	err := c.WriteToFile(tempFile("styled_paragraph_table_vertical_align.pdf"))
```
Comment (gunnsth) [inline creator/styled_paragraph_test.go]: Would be good to add a rendering test on this too, to catch any regression related to this (call testWriteAndRender).

File: creator/styled_paragraph.go [hunk 2]
```diff
@@ -226,6 +226,48 @@ func (p *StyledParagraph) Height() float64 {
...
 }
 
+// getLineHeight returns both the capheight and the font size based height of
+// the line with the specified index.
+func (p *StyledParagraph) getLineHeight(idx int) (capHeight, height float64) {
+	if p.lines == nil || len(p.lines) == 0 {
+		p.wrapText()
+	}
+	if idx < 0 || idx > len(p.lines)-1 {
+		return 0, 0
```
Comment (gunnsth) [inline creator/styled_paragraph.go]: Output a debug message that it is out of range.

Comment (adrg) [reply creator/styled_paragraph.go]: Done

File: creator/styled_paragraph_test.go (hunk 1 above)
Comment (adrg) [reply creator/styled_paragraph_test.go]: Done

**Pull request merged**